backup/backup-disks.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/python3
+
+""" Primary Backup Logic"""
+import json
+import sys
+import datetime
+
+from datetime import date
+from settings import settings
+from googleapiclient import discovery
+from oauth2client.client import GoogleCredentials
+from json.decoder import JSONDecodeError as JsonError
+
+
+def list_disks(compute, project, zone):
+    """ Lists all persistent disks used by project """
+    all_disks = []
+    result = compute.disks().list(project=project, zone=zone).execute()
+    all_disks.extend(result['items'])
+
+    while 'nextPageToken' in result:
+        result = compute.disks().list(project=project, zone=zone, \
+            pageToken=result['nextPageToken']).execute()
+        all_disks.extend(result['items'])
+
+    return all_disks
+
+
+def list_snapshots(compute, project):
+    """ Lists all snapshots created for this project """
+    all_snapshots = []
+    result = compute.snapshots().list(project=project).execute()
+    all_snapshots.extend(result['items'])
+
+    while 'nextPageToken' in result:
+        result = compute.snapshots().list(project=project, \
+            pageToken=result['nextPageToken']).execute()
+        all_snapshots.extend(result['items'])
+
+    return all_snapshots
+
+
+def filter_disks_by_name(disks, name):
+    """ Takes in NAME, a predefined prefix to disk names, and filters 
+    disks to only returns those that have NAME """
+    filtered_disks = []
+    for disk in disks:
+        try:
+            if name in disk['name']:
+                filtered_disks.append(disk)
+        except KeyError:
+            sys.exit("Improperly formatted disks -- is your information correct?")
+    return filtered_disks
+
+
+def filter_disks_by_namespace(disks, namespace, namespace_dict_key):
```

Comment:
Is this method here to allow future functionality that allows users to specify which namespace they want to create a snapshot of?